### PR TITLE
Stellar federated address in profile

### PR DESCRIPTION
**Pull request by zanderz**
mobile and desktop. Need some help on the toast popup on copy though.
screen shot 2018-11-20 at 4 20 17 pm

Also, how do you make a divider in the floating menu on mobile?
screen shot 2018-11-20 at 4 19 52 pm
screen shot 2018-11-20 at 4 29 21 pm

### CODE REVIEW

**Comment by chrisnojima** [comment]
These should show only on users with wallets

**Comment by zanderz** [comment]
oh I just made a fake one show up in the __DEV__ case for now, since the federated address is supposed to come from the service and isn't done yet: https://keybase.atlassian.net/browse/CORE-9435

File: shared/profile/container.js [hunk 1]
```diff
@@ -178,6 +186,7 @@ const mergeProps = (stateProps, dispatchProps) => {
     onBack: stateProps.profileIsRoot ? null : dispatchProps.onBack,
     onBrowsePublicFolder: () => dispatchProps._onBrowsePublicFolder(username),
     onChat: () => dispatchProps._onChat(username),
+    onCopyStellarAddress,
```
Comment (chrisnojima) [inline shared/profile/container.js]: just inline it like the rest of the functions here

File: shared/profile/user-actions.js [hunk 2]
```diff
@@ -194,17 +197,93 @@ const _DropdownButton = (props: DropdownProps & OverlayParentProps) => (
...
 
-const ellipsisIcon = platformStyles({
+export type StellarFederatedAddressProps = {|
+  currentlyFollowing?: boolean,
+  stellarAddress: string,
+  onSendOrRequest: (isRequest: boolean) => void,
+  onCopyAddress: () => void,
+|}
+
+export const MakeStellarAddressMenuItems = (props: StellarFederatedAddressProps) => [
```
Comment (chrisnojima) [inline shared/profile/user-actions.js]: this isn't a component, so lets not uppercase it

File: shared/profile/index.native.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ import Folders from './folders/container'
 import React, {Component} from 'react'
 import {chunk} from 'lodash-es'
 import moment from 'moment'
-import UserActions from './user-actions'
+import UserActions, {MakeStellarAddressMenuItems, type StellarFederatedAddressProps} from './user-actions'
```
Comment (chrisnojima) [inline shared/profile/index.native.js]: see comment below about casing

Comment (zanderz) [reply shared/profile/index.native.js]: ok sounds good

File: shared/profile/user-actions.js (hunk 2 above)
Comment (zanderz) [reply shared/profile/user-actions.js]: ok

File: shared/profile/index.desktop.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ import Friendships from './friendships.desktop'
 import * as React from 'react'
 import moment from 'moment'
 import * as Kb from '../common-adapters'
-import UserActions, {MakeStellarAddressMenuItems, type StellarFederatedAddressProps} from './user-actions'
+import UserActions, {makeStellarAddressMenuItems, type stellarFederatedAddressProps} from './user-actions'
```
Comment (chrisnojima) [inline shared/profile/index.desktop.js]: the type should be capitolized

Comment (zanderz) [reply shared/profile/index.desktop.js]: ok

**Comment by akalin-keybase** [comment]
I'll review this, sorry for the delay

File: shared/profile/index.native.js [hunk 5]
```diff
@@ -340,6 +381,14 @@ class Profile extends Component<Props, State> {
             loadingComponent={this._makeUserProofs(true)}
             doneLoadingComponent={this._makeUserProofs(false)}
           />
+          {this.props.stellarAddress && !this.props.loading && (
```
[the remark this answers is not in the document]
Comment (zanderz) [reply shared/profile/index.native.js]: ok

File: shared/profile/index.desktop.js [hunk 6]
```diff
@@ -407,15 +495,23 @@ class ProfileRender extends React.PureComponent<Props, State> {
                     showingMenuIndex={this.state.selectedProofMenuRowIndex}
                   />
                 )}
+                {this.props.stellarAddress && !loading && (
```
[the remark this answers is not in the document]
Comment (zanderz) [reply shared/profile/index.desktop.js]: ok

File: shared/profile/index.desktop.js [hunk 7]
```diff
@@ -83,26 +68,129 @@ const _ShowcasedTeamRow = (
...
+      <Kb.Toast position="top left" attachTo={this.props.getAttachmentRef} visible={this.state.showingToast}>
+        {Styles.isMobile && <Kb.Icon type="iconfont-clipboard" color="white" fontSize={22} />}
+        <Kb.Text type={Styles.isMobile ? 'BodySmallSemibold' : 'BodySmall'} style={styles.toastText}>
+          Copied to clipboard
+        </Kb.Text>
+      </Kb.Toast>
+    )
+  }
+}
+const ToastContainer = HOCTimers(_ToastContainer)
```
[the remark this answers is not in the document]
Comment (zanderz) [reply shared/profile/index.desktop.js]: sounds good

**Comment by zanderz** [comment]
thanks @akalin , did those things

**Pull request merged**